### PR TITLE
Improve documentation for PyPREP/MatPREP highpass differences

### LINKED ISSUE #107: filt vs filtfilt (MATLAB implementation)
@sappelhoff So I'm currently taking a grad course on signal processing, and I think I finally understand the key difference between MNE's highpass filter and EEGLAB's: MNE defaults to applying the filter using a zero-phase method that doesn't result in any phase shift in the signal (the equivalent of `filtfilt` in MATLAB/Scipy), whereas the EEGLAB filter function I re-implemented here only does a single forward pass (using `lfilter`), so you end up with phase shift in the signal.

As far as I can tell, other than theoretically being faster (which it isn't in the current implementation) there's absolutely no reason to use a forwards-only filter like EEGLAB does instead of the backwards-forwards zero-phase approach used by MNE. I'm guessing this is a weird legacy carry-over for EEGLAB from the days when people were running EEG analyses on Pentium 3's and PowerPC G4's so CPU cycles were at a major premium. I'll update the "differences" docs to reflect this when I get a chance!

_Originally posted by @a-hurst in https://github.com/sappelhoff/pyprep/issues/71#issuecomment-973060655_

### PR DESCRIPTION
**Pull request by a-hurst**
# PR Description

Closes #107. This PR updates the docs to contain a better explanation of the differences between PyPREP's and MatPREP's trend removal filters (and why the MNE method is technically preferable).

Wasn't sure if this warranted an addition in the `whats_new.rst`, but if it does I can definitely add a line!

# Merge Checklist

- [x] the PR has been reviewed and all comments are resolved
- [x] all [CI][what-is-ci] checks pass
- [x] (if applicable): the PR description includes the phrase `closes #<issue-number>` to [automatically close an issue][auto-close-documentation]
- [ ] (if applicable): bug fixes, new features, or [API][what-is-api] changes are documented in [whats_new.rst][whats-new-file]

[what-is-ci]: https://help.github.com/en/actions/building-and-testing-code-with-continuous-integration/about-continuous-integration
[auto-close-documentation]: https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
[what-is-api]: https://en.wikipedia.org/wiki/Application_programming_interface
[whats-new-file]: https://github.com/sappelhoff/pyprep/blob/master/docs/whats_new.rst

### CODE REVIEW

**Review by sappelhoff — APPROVED** [review]
perfect :-) Thanks a lot!

**Comment by a-hurst** [comment]
Er, looks like MNE needs an additional module to download test datasets now? Should I add that to this PR as well?

**Comment by sappelhoff** [comment]
ah right :thinking: yes please - we can just add it here: https://github.com/sappelhoff/pyprep/blob/master/requirements-dev.txt

**Comment by sappelhoff** [comment]
although `pooch` may soon be installed automatically when we install `mne`, see: https://github.com/mne-tools/mne-python/pull/10199

but no harm in adding it now to merge this PR, and then removing it again some time in the future when we don't need to declare it explicitly anymore.

**Comment by a-hurst** [comment]
Whoops, looks like matplotlib's changed its API on us since last update too. I'll try to tackle that as well.

File: requirements-dev.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@ flake8
 flake8-docstrings
 isort
 matplotlib
+mne[data]
```
Comment (sappelhoff) [inline requirements-dev.txt:7]: I prefer to be explicit here:

```suggestion
tqdm
pooch
```

Comment (a-hurst) [reply requirements-dev.txt:7]: Ah, okay. I figured `mne[data]` would future-proof against future MNE changes to data downloading, but being explicit manse sense.

**Comment by sappelhoff** [comment]
Thank you :)

**Pull request merged**